lib/provisioning/ProvisionDispatcher.js

```diff
@@ -0,0 +1,356 @@
+                           { zkPath: this.zkEndpoint + zkPath });
+            return cb();
+        });
+    }
+
+    _amILeader(children) {
+        return children.every(child => child >= this.myLeaderName);
+    }
+
+    _onLeaderChange() {
+        // monitor leader change
+        const zkPath = ProvisionDispatcher.constants.LEADERS;
+        this.client.getChildren(zkPath, event => {
+            this.log.debug('got leader event',
+                           { zkPath: this.zkEndpoint + zkPath,
+                             event });
+            this._onLeaderChange();
+        }, (err, children) => {
+            if (err) {
+                this.log.error('list failed',
+                               { zkPath: this.zkEndpoint + zkPath,
+                                 error: err });
+                return;
+            }
+            this.isLeader = this._amILeader(children);
+            this.log.debug('updated leader',
+                           { zkPath: this.zkEndpoint + zkPath,
+                             children,
+                             iamLeader: this.isLeader });
+            if (this.isLeader) {
+                this._redispatchProvisions();
+            }
+        });
+    }
+
+    _onOwnerChange() {
+        const zkPath = ProvisionDispatcher.constants.OWNERS;
+        this.client.getChildren(zkPath, event => {
+            this.log.debug('got owner event',
+                           { zkPath: this.zkEndpoint + zkPath,
+                             event });
+            this._onOwnerChange();
+        }, (err, children) => {
+            if (err) {
+                this.log.error('list failed',
+                               { zkPath: this.zkEndpoint + zkPath,
+                                 error: err });
+                return;
+            }
+            this.owners = children;
+            this.log.debug('owners updated',
+                           { zkPath: this.zkEndpoint + zkPath,
+                             owners: this.owners });
+            if (this.isLeader) {
+                this._redispatchProvisions();
+            }
+        });
+    }
+
+    _onMyselfChange(cb) {
+        // monitor change in my content
+        const myPath = `${ProvisionDispatcher.constants.OWNERS}/` +
+                  `${this.myName}`;
+        this.client.getData(
+            myPath, event => {
+                this.log.debug('got owner self event',
+                               { zkPath: this.zkEndpoint + myPath,
+                                 event });
+                this._onMyselfChange(cb);
+            },
+            (err, data) => {
+                if (err &&
+                    err.getCode() !== zookeeper.Exception.NO_NODE) {
+                    this.log.error('error in getData',
+                                   { zkPath: this.zkEndpoint + myPath,
+                                     error: err });
+                    return undefined;
```

Comment:
We can, it's not a strong functional requirement but it would allow to notify the caller of errors so that he can log them as well as here (or do some other action). I will do the change.